lib/compress/zstdmt_compress.c

```diff
@@ -263,6 +261,16 @@ static void ZSTDMT_releaseBuffer(ZSTDMT_bufferPool* bufPool, buffer_t buf)
     ZSTD_customFree(buf.start, bufPool->cMem);
 }
 
+/* We need 2 output buffers per worker since each dstBuff must be flushed after it is released.
+ * The 3 additional buffers are as follows:
+ *   1 buffer for input loading
+ *   1 buffer for "next input" when submitting current one
+ *   1 buffer stuck in queue */
+#define BUF_POOL_MAX_NB_BUFFERS(nbWorkers) 2*nbWorkers + 3
```

Comment:
Hmmm. This is potentially dangerous. Macro expressions and parameters should always be carefully wrapped (either with parentheses or a `do { ... } while (0)` as appropriate). A safe way to write this would be `(2 * (nbWorkers) + 3)`. Without the parentheses, using this macro like `BUF_POOL_MAX_NB_BUFFERS(foo) * 2` or `BUF_POOL_MAX_NB_BUFFERS(foo + 1)` will produce incorrect results. 

You're only using it directly as an argument to functions, with a direct argument, so this PR avoids hitting these landmines. But it may still be worth fixing?

Reply:
Good catch, will put up a fix

Reply:
We don't need `do {...} while(0)` in this case, as it's just a mathematical expression and we need it to return a value,
but definitely, enclosing the variable inside brackets `(nbWorkers)` is a good code-hardening practice.